src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7743,11 +7762,16 @@ private ForStatementSyntax ParseForStatement()
                 }
                 else
                 {
-                    isDeclaration = !this.IsQueryExpression(mayBeVariableDeclaration: true, mayBeMemberDeclaration: false) &&
-                                    this.ScanType() != ScanTypeFlags.NotType &&
-                                    this.IsTrueIdentifier();
+                    decl = TryParseDeconstructionDeclaration(withEquals: true);
 
-                    this.Reset(ref resetPoint);
+                    if (decl == null)
+                    {
+                        isDeclaration = !this.IsQueryExpression(mayBeVariableDeclaration: true, mayBeMemberDeclaration: false) &&
+                                        this.ScanType() != ScanTypeFlags.NotType &&
+                                        this.IsTrueIdentifier();
+
+                        this.Reset(ref resetPoint);
+                    }
```

Comment:
No, I meant the situation when we have some things after the deconstruction declaration but before the semicolon.

---
In reply to: [68294054](https://github.com/dotnet/roslyn/pull/12104#discussion_r68294054) [](ancestors = 68294054)